examples/microwave-oven-app/microwave-oven-common/src/microwave-oven-device.cpp

```diff
@@ -35,22 +35,55 @@ void ExampleMicrowaveOvenDevice::MicrowaveOvenInit()
     mOperationalStateInstance.Init();
     mMicrowaveOvenModeInstance.Init();
     mMicrowaveOvenControlInstance.Init();
+
+    // set default value for attribute SelectedWattIndex and WattRating
+    if (mMicrowaveOvenControlInstance.HasFeature(MicrowaveOvenControl::Feature::kPowerInWatts))
+    {
+        static_assert(ArraySize(mWattSettingList) != 0, "Watt setting list is empty!");
+        mSelectedWattIndex = ArraySize(mWattSettingList) - 1;
+        mWattRating        = mWattSettingList[mSelectedWattIndex];
+    }
+    else
+    {
+        mWattRating = kExampleWatt5;
+    }
 }
 
 /**
  * MicrowaveOvenControl cluster
  */
 Protocols::InteractionModel::Status
-ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, uint8_t powerSetting)
+ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, bool startAfterSetting,
+                                                               Optional<uint8_t> powerSetting, Optional<uint8_t> wattSettingIndex)
 {
     // placeholder implementation
     Status status;
+    // set cook mode
     if ((status = mMicrowaveOvenModeInstance.UpdateCurrentMode(cookMode)) != Status::Success)
     {
         return status;
     }
+
+    // set cook time
```

Comment:
comment does not seem to increase understanding of the code - if I read `SetCookTime` being called, I assume that the cook time is being set. Same applies to several other comments below. Please rephrase to explain things non-obvious in the code (if there are ordering issues or specific values or explain when some things will not be set) or remove.